test2/task1/main.c

```diff
@@ -0,0 +1,162 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+char *maxDataFunction(char *data1, char *data2)
+{
+    // year
+    int index1 = 6;
+    int index2 = 6;
+    while (data1[index1] != '\0')
+    {
+        ++index1;
+    }
+    while (data2[index2] != '\0')
+    {
+        ++index2;
+    }
+
+    if (index1 > index2)
+    {
+        return data1;
+    }
+    if (index1 < index2)
+    {
+        return data2;
+    }
+
+    int index = 6;
+
+    for (;;)
+    {
+        if (data1[index] > data2[index])
+        {
+            return data1;
+        }
+        if (data1[index] < data2[index])
+        {
+            return data2;
+        }
+        ++index;
+    }
+
+    // month
+    if (data1[3] > data2[3])
+    {
+        return data1;
+    }
+    if (data1[3] < data2[3])
+    {
+        return data2;
+    }
+    if (data1[4] > data2[4])
+    {
+        return data1;
+    }
+    if (data1[4] < data2[4])
+    {
+        return data2;
+    }
+
+    // day
+    if (data1[0] > data2[0])
+    {
+        return data1;
+    }
+    if (data1[0] < data2[0])
+    {
+        return data2;
+    }
+    if (data1[1] > data2[1])
+    {
+        return data1;
+    }
+    if (data1[1] < data2[1])
+    {
+        return data2;
+    }
+
+    // if equal
+    return data1;
+}
+
+int main()
+{
+    FILE *file = fopen("text.txt", "r");
+
+    bool isDataInText = false;
+    char currentSign;
+
+    char *data = calloc(15, sizeof(char));
```

Comment:
```suggestion
    char *date = calloc(15, sizeof(char));
```
date - дата
data - данные

Reply:
А проверить, что память успешно выделена?
Укажу только здесь, но в программе есть еще такие же ошибки.
(-0.5 балла)